src/Symfony/Component/ExpressionLanguage/ParsedExpression.php

```diff
@@ -42,6 +42,17 @@ public function getNodes()
 
     public function dump()
     {
-        return $this->nodes->dump();
+        return $this->dumpNode($this->nodes);
+    }
+
+    protected function dumpNode(Node $node)
```

Comment:
IMHO, the extension point is not really easy to use. Because the EL returns a `ParsedExpression` and not a `ParsedExpressionToDumpInAnHTMLShape` you have to do some ugly code.